### PR TITLE
Add `start` and `end` args to ViewTool

### PR DESCRIPTION
**Pull request by matheper**
This pull request enhances the `ViewTool` functionality in the `debug_gym` project by introducing support for specifying line ranges when viewing files. It also improves the `show_line_number` utility to handle large files and custom starting indices, and adds comprehensive tests to ensure correctness. Below are the most significant changes:

### Enhancements to `ViewTool` functionality:
* Added `start` and `end` parameters to the `ViewTool` schema, allowing users to specify a range of lines (inclusive) to view in a file. Updated the `use` method to validate and handle these parameters, including checks for invalid ranges and indices. (`debug_gym/gym/tools/view.py`) [[1]](diffhunk://#diff-0d3bd38bfe264b7cf3162068122588906b89dc8ef812190540aefb2651de13bcR28-R35) [[2]](diffhunk://#diff-0d3bd38bfe264b7cf3162068122588906b89dc8ef812190540aefb2651de13bcR46-L77)

### Improvements to `show_line_number` utility:
* Modified the `show_line_number` function to support a `start_index` parameter, enabling custom starting line numbers. Enhanced the logic for formatting line numbers and handling edge cases, such as empty code strings. (`debug_gym/gym/utils.py`)

### Test suite expansion:
* Refactored and added new test cases for `show_line_number`, including tests for empty code strings, custom starting indices, and large files with thousands of lines. (`tests/gym/test_utils.py`) [[1]](diffhunk://#diff-8fe9b7007890dd34347c5caf973a6d362105712372ec557125e3303b8e453528L34-R43) [[2]](diffhunk://#diff-8fe9b7007890dd34347c5caf973a6d362105712372ec557125e3303b8e453528L53-R73) [[3]](diffhunk://#diff-8fe9b7007890dd34347c5caf973a6d362105712372ec557125e3303b8e453528L84-R162)
* Introduced tests for the updated `ViewTool` to validate line range functionality, including edge cases like full content, partial ranges, and invalid ranges. (`tests/gym/tools/test_view.py`)

### Minor cleanup:
* Removed unused imports in `debug_gym/gym/envs/env.py`. (`debug_gym/gym/envs/env.py`) 

### Test data addition:
* Added a new test file, `ten_lines.py`, to the test environment setup for validating line range functionality in `ViewTool`. (`tests/gym/tools/test_view.py`)

### CODE REVIEW

File: debug_gym/gym/tools/view.py [hunk 1]
```diff
@@ -25,6 +25,14 @@ class ViewTool(EnvironmentTool):
             "type": ["string"],
             "description": "The path to the file to be viewed. The path should be relative to the root directory of the repository.",
         },
+        "start": {
+            "type": ["integer", "null"],
```
Comment (xingdi-eric-yuan) [inline debug_gym/gym/tools/view.py]: "integer" or "number"?

File: debug_gym/gym/tools/view.py [hunk 2]
```diff
@@ -35,44 +43,79 @@ def use(
...
+                start_index=s + 1,
+            )
+            if environment.current_breakpoints_state:
+                breakpoints_message = (
+                    ". B indicates breakpoint before a certain line of code"
+                )
 
+        read_only = " (read-only)" if not environment.is_editable(new_file) else ""
+        obs = (
+            f"Viewing `{new_file}`{read_only}{breakpoints_message}:"
```
Comment (xingdi-eric-yuan) [inline debug_gym/gym/tools/view.py]: We can mention the start and end line numbers in ths message. 
Because when `include_line_numbers_and_breakpoints is False` the range info is missing.

File: debug_gym/gym/tools/view.py (hunk 1 above)
Comment (xingdi-eric-yuan) [reply debug_gym/gym/tools/view.py]: reference:
https://community.openai.com/t/types-accepted-by-call-functions/549169/6
